schema/validator.go

```diff
@@ -54,6 +55,28 @@ func (v Validator) Validate(src io.Reader) error {
 			"schema %s: unable to validate", v)
 	}
 
+	if v == MediaTypeManifest {
```

Comment:
On Wed, Sep 07, 2016 at 04:00:53AM -0700, xiekeyang wrote:
> > +	if v == MediaTypeManifest {
>
> @wking `validate --type manifest` type seems not to jump into
> `manifest.validate`, instead into `Validator.Validate` directly. So,
> `manifest.validate` can't be effective.

?  ‘oci-image-tool validate …’ calls both image.Validate(Layout) [1]
and schema.MediaType(Manifest|ManifestList|Config).Validate [2].  I
expect we should be calling the schema validation from the image
validation functions, but either way, code in the image validation
functions is getting called by ‘oci-image-tool validate …’.

[1]: https://github.com/opencontainers/image-spec/blob/d3cfb991ee3ff9e7c9570027cb9696213cfcee00/cmd/oci-image-tool/validate.go#L126-L131
[2]: https://github.com/opencontainers/image-spec/blob/d3cfb991ee3ff9e7c9570027cb9696213cfcee00/cmd/oci-image-tool/validate.go#L139-L146